panic: could not determine route type

### Naabu version: 2.3.0

### Current Behavior:
Panics on start.

% ./naabu -v
panic: could not determine route type for: 'default            link#38            UCSg            utun8'

goroutine 1 [running]:
github.com/projectdiscovery/naabu/v2/pkg/scan.init.2()
	/Users/runner/work/naabu/naabu/v2/pkg/scan/scan_common.go:63 +0x70

### Expected Behavior:
No panic. As documented.

### Steps To Reproduce:
OS: MacOS Ventura 13.2.1
VPN is present.

--- a/v2/pkg/runner/options.go
+++ b/v2/pkg/runner/options.go
@@ -270,7 +270,7 @@ func (options *Options) ShouldLoadResume() bool {
 }
 
 func (options *Options) shouldDiscoverHosts() bool {
-	return (options.OnlyHostDiscovery || !options.SkipHostDiscovery) && !options.Passive
+	return (options.OnlyHostDiscovery || !options.SkipHostDiscovery) && !options.Passive && scan.PkgRouter != nil
 }
 
 func (options *Options) hasProbes() bool {
@@ -280,5 +280,5 @@ func (options *Options) hasProbes() bool {
 }
 
 func (options *Options) shouldUseRawPackets() bool {
-	return isOSSupported() && privileges.IsPrivileged && options.ScanType == SynScan
+	return isOSSupported() && privileges.IsPrivileged && options.ScanType == SynScan && scan.PkgRouter != nil
 }

--- a/v2/pkg/runner/validate.go
+++ b/v2/pkg/runner/validate.go
@@ -139,6 +139,11 @@ func (options *Options) ValidateOptions() error {
 		options.ScanType = ConnectScan
 	}
 
+	if options.ScanType == SynScan && scan.PkgRouter == nil {
+		gologger.Warning().Msgf("Routing could not be determined (are you using a VPN?).falling back to connect scan")
+		options.ScanType = ConnectScan
+	}
+
 	return nil
 }
 

--- a/v2/pkg/scan/arp_unix.go
+++ b/v2/pkg/scan/arp_unix.go
@@ -17,7 +17,7 @@ func init() {
 
 // ArpRequestAsync asynchronous to the target ip address
 func arpRequestAsync(ip string) {
-	networkInterface, _, sourceIP, err := pkgRouter.Route(net.ParseIP(ip))
+	networkInterface, _, sourceIP, err := PkgRouter.Route(net.ParseIP(ip))
 	if networkInterface == nil {
 		err = errors.New("Could not send ARP Request packet to " + ip + ": no interface with outbound source found")
 	}

--- a/v2/pkg/scan/icmp.go
+++ b/v2/pkg/scan/icmp.go
@@ -95,7 +95,7 @@ func PingIcmpEchoRequestAsync(ip string) {
 	case iputil.IsIPv6(ip):
 		m.Type = ipv6.ICMPTypeEchoRequest
 		packetListener = icmpConn6
-		networkInterface, _, _, err := pkgRouter.Route(destinationIP)
+		networkInterface, _, _, err := PkgRouter.Route(destinationIP)
 		if networkInterface == nil {
 			err = fmt.Errorf("could not send ICMP Echo Request packet to %s: no interface with outbout source ipv6 found", destinationIP)
 		}

--- a/v2/pkg/scan/ndp.go
+++ b/v2/pkg/scan/ndp.go
@@ -15,7 +15,7 @@ import (
 
 // PingNdpRequestAsync asynchronous to the target ip address
 func PingNdpRequestAsync(ip string) {
-	networkInterface, _, _, err := pkgRouter.Route(net.ParseIP(ip))
+	networkInterface, _, _, err := PkgRouter.Route(net.ParseIP(ip))
 	if networkInterface == nil {
 		err = errors.New("Could not send PingNdp Request packet to " + ip + ": no interface with outbound source found")
 	}

--- a/v2/pkg/scan/scan_common.go
+++ b/v2/pkg/scan/scan_common.go
@@ -4,6 +4,7 @@ import (
 	"errors"
 	"net"
 
+	"github.com/projectdiscovery/gologger"
 	"github.com/projectdiscovery/naabu/v2/pkg/privileges"
 	"github.com/projectdiscovery/naabu/v2/pkg/routing"
 	"golang.org/x/net/icmp"
@@ -21,7 +22,7 @@ var (
 	transportPacketSend, icmpPacketSend, ethernetPacketSend chan *PkgSend
 	icmpConn4, icmpConn6                                    *icmp.PacketConn
 
-	pkgRouter routing.Router
+	PkgRouter routing.Router
 
 	ArpRequestAsync  func(ip string)
 	InitScanner      func(s *Scanner) error
@@ -39,7 +40,7 @@ type ListenHandler struct {
 
 func Acquire() (*ListenHandler, error) {
 	// always grant to unprivileged scans
-	if !privileges.IsPrivileged {
+	if PkgRouter == nil || !privileges.IsPrivileged {
 		return &ListenHandler{Phase: &Phase{}}, nil
 	}
 
@@ -60,9 +61,9 @@ func (l *ListenHandler) Release() {
 
 func init() {
 	if r, err := routing.New(); err != nil {
-		panic(err)
+		gologger.Error().Msgf("could not initialize router: %s\n", err)
 	} else {
-		pkgRouter = r
+		PkgRouter = r
 	}
 }
 

--- a/v2/pkg/scan/scan_unix.go
+++ b/v2/pkg/scan/scan_unix.go
@@ -39,7 +39,7 @@ type Handlers struct {
 }
 
 func init() {
-	if !privileges.IsPrivileged {
+	if PkgRouter == nil || !privileges.IsPrivileged {
 		return
 	}
 
@@ -170,7 +170,7 @@ func sendAsyncTCP4(listenHandler *ListenHandler, ip string, p *port.Port, pkgFla
 		TTL:      255,
 		Protocol: layers.IPProtocolTCP,
 	}
-	_, _, sourceIP, err := pkgRouter.Route(ip4.DstIP)
+	_, _, sourceIP, err := PkgRouter.Route(ip4.DstIP)
 	if err != nil {
 		gologger.Debug().Msgf("could not find route to host %s:%d: %s\n", ip, p.Port, err)
 		return
@@ -219,7 +219,7 @@ func sendAsyncUDP4(listenHandler *ListenHandler, ip string, p *port.Port, pkgFla
 		TTL:      255,
 		Protocol: layers.IPProtocolUDP,
 	}
-	_, _, sourceIP, err := pkgRouter.Route(ip4.DstIP)
+	_, _, sourceIP, err := PkgRouter.Route(ip4.DstIP)
 	if err != nil {
 		gologger.Debug().Msgf("could not find route to host %s:%d: %s\n", ip, p.Port, err)
 		return
@@ -254,7 +254,7 @@ func sendAsyncTCP6(listenHandler *ListenHandler, ip string, p *port.Port, pkgFla
 		NextHeader: layers.IPProtocolTCP,
 	}
 
-	_, _, sourceIP, err := pkgRouter.Route(ip6.DstIP)
+	_, _, sourceIP, err := PkgRouter.Route(ip6.DstIP)
 	if err != nil {
 		gologger.Debug().Msgf("could not find route to host %s:%d: %s\n", ip, p.Port, err)
 		return
@@ -304,7 +304,7 @@ func sendAsyncUDP6(listenHandler *ListenHandler, ip string, p *port.Port, pkgFla
 		NextHeader: layers.IPProtocolUDP,
 	}
 
-	_, _, sourceIP, err := pkgRouter.Route(ip6.DstIP)
+	_, _, sourceIP, err := PkgRouter.Route(ip6.DstIP)
 	if err != nil {
 		gologger.Debug().Msgf("could not find route to host %s:%d: %s\n", ip, p.Port, err)
 		return
@@ -785,7 +785,7 @@ func ACKPort(listenHandler *ListenHandler, dstIP string, port int, timeout time.
 		Protocol: layers.IPProtocolTCP,
 	}
 
-	_, _, sourceIP, err := pkgRouter.Route(ip4.DstIP)
+	_, _, sourceIP, err := PkgRouter.Route(ip4.DstIP)
 	if err != nil {
 		return false, err
 	}

